### DISCUSSION TITLE
Api request with related tables

### DISCUSSION THREAD

**Discussion by fellowshipdevs** [supabase/supabase#3530] [Questions] [answered]
Here's my use case:

I have a table of "items", which has an int column of "status_id". This "state_id" column is a link to a referential table "item_statuses". This "item_statusess" has 2 columns: (int) "id" and (varchar) "status_label". My question is, how can I retrieve the "status_label" when querying the "items" table via the API? Is this even possible?

Thanks.

**Comment by steve-chavez** [comment, marked as answer]
>  Is this even possible?

Yes, see https://supabase.io/docs/reference/javascript/select#query-foreign-tables. Should be like:

```js
const { data, error } = await supabase
  .from('items')
  .select(`
    *,
    item_statuses(status_label)
  `)
```

**Reply by fellowshipdevs** [reply]
Thanks, but that's when using javascript. I am using the auto-generated table API endpoints, e.g. https://vvrsdasdfsrrakghz.supabase.co/rest/v1/items?select=*. Unless there's a way to marry the 2 (javascript and API)?

**Reply by fellowshipdevs** [reply]
Oh, nvm, I think I found it in the documentation. My bad.

**Reply by VictorPeralta** [reply]
For future references, this is the postgrest documentation where this is specified

https://postgrest.org/en/v8.0/api.html#resource-embedding
